collect_app/src/main/java/org/odk/collect/android/backgroundwork/SyncFormsTaskSpec.kt

```diff
@@ -18,9 +18,12 @@ class SyncFormsTaskSpec : TaskSpec {
     override fun getTask(context: Context, inputData: Map<String, String>): Supplier<Boolean> {
         DaggerUtils.getComponent(context).inject(this)
         return Supplier {
-            formsUpdater.matchFormsWithServer(
-                inputData[TaskSpec.DATA_PROJECT_ID]!!
-            )
+            val projectId = inputData[TaskSpec.DATA_PROJECT_ID]
+            if (projectId != null) {
+                return@Supplier formsUpdater.matchFormsWithServer(projectId)
```

Comment:
You can just omit the `return` here so you don't need the whole `@Supplier` thing.

Reply:
Right, fixed.